### DISCUSSION TITLE
Minimal Memory Usage

### DISCUSSION THREAD

**Discussion by cfengno1** [block-hczhai/block2-preview#151] [Q&A] [answered]
Hi Huanchen,

I need to do large scale DMRG simulations of model systems, so memory efficiency is important. The Hamiltonian MPO is not a big deal. I am wondering if the default setup in block2 is already optimal, or are there parameters that can save memory usage?

Thank you very much!

Best,
Feng

**Comment by hczhai** [comment, marked as answer]
The optimal settings can be system dependent. Generic ways to reduce the memory usage can be:

1. Set ``symm_type=SymmetryTypes.SU2 | SymmetryTypes.SP`` to activate single precision mode. This should work for most model systems, and reduce half of the memory and computational cost. The Davidson threshold ``thrds`` should be set to 1E-5 to 1E-6 in this mode.
2. Set ``dot=1`` in ``driver.get_random_mps`` to do 1-site DMRG. It may converge to local minima in some cases.
3. Use MPI and multiple nodes to distribute the memory. To get instructions on how to do this I have to know the complete input file.

If the above does not solve the problem, further system specific suggestions have to be based on the detailed information on the system, the resource constraint, input and output file, and the current and expected performance.
